helix-term/src/commands/lsp.rs

```diff
@@ -452,43 +452,73 @@ pub fn code_action(cx: &mut Context) {
     cx.callback(
         future,
         move |editor, compositor, response: Option<lsp::CodeActionResponse>| {
-            let actions = match response {
+            let mut actions = match response {
                 Some(a) => a,
                 None => return,
             };
+
             if actions.is_empty() {
                 editor.set_status("No code actions available");
                 return;
             }
 
-            let mut picker = ui::Menu::new(actions, (), move |editor, code_action, event| {
-                if event != PromptEvent::Validate {
-                    return;
+            // sort by CodeActionKind
+            // this ensures that the most relevant codeactions (quickfix) show up first
+            // while more situational commands (like refactors) show up later
+            // this behaviour is modeled after the behaviour of vscode (editor/contrib/codeAction/browser/codeActionWidget.ts)
```

Comment:
Could this be a permalink instead? So the reference doesn't get lost if vscode reorganizes their repo